nixos/modules/system/activation/switch-to-configuration.pl

```diff
@@ -889,9 +889,16 @@ sub filter_units {
 
 close($list_active_users) || die("Unable to close the file handle to loginctl");
 
-# Set the new tmpfiles
-print STDERR "setting up tmpfiles\n";
-system("$new_systemd/bin/systemd-tmpfiles", "--create", "--remove", "--exclude-prefix=/dev") == 0 or $res = 3;
+# Restart sysinit-reactivation.target.
+# This target only exists to restart services ordered before sysinit.target. We
+# cannot use X-StopOnReconfiguration to restart sysinit.target because then ALL
+# services of the system would be restarted since all normal services have a
+# default dependency on sysinit.target. This target ensures that services
+# ordered before sysinit.target get started in the correct order, most notably
+# BEFORE the rest of the services. Ordering between these services is
+# respected.
+print STDERR "restarting sysinit-reactivation.target\n";
+system("$new_systemd/bin/systemctl", "restart", "sysinit-reactivation.target") == 0 or $res = 4;
```

Comment:
Can we also use this than for agenix and sops-nix?

Reply:
This gives us better visibility when monitoring a service I would say. One can export service state as metrics and also export logs.

Reply:
No the big argument is that we can now respect the ordering between the services pulled in by sysinit. Concretely, this allows us to restart sysusers BEFORE tmpfiles which wasn't previously possible.